avocado/plugins/runner_nrunner.py

```diff
@@ -359,8 +359,7 @@ def run_suite(self, job, test_suite):
                     if rt.task.category == 'test']
         tsm = TaskStateMachine(self.runtime_tasks, self.status_repo)
         spawner_name = test_suite.config.get('nrunner.spawner')
-        spawner = SpawnerDispatcher(test_suite.config)[spawner_name].obj
-        spawner.job_output_dir = job.test_results_path
+        spawner = SpawnerDispatcher(test_suite.config, job)[spawner_name].obj
```

Comment:
Indeed, I didn't notice that you have now added the job as argument, in my case I was doing:
```python
self.spawner = SpawnerDispatcher(job.config)[spawner_name].obj
```
and will currently have to change it to
```python
self.spawner = SpawnerDispatcher(job.config, job)[spawner_name].obj
```
which is a bit strange at least from my current perspective (we only pass around and use the job-level config) but that's fine and I will try this now.

Reply:
Indeed, this did the job (pun intended).